### DISCUSSION TITLE
Completion returns either a 404 or 500 server error

### DISCUSSION THREAD

**Discussion by jkirkcaldy** [open-webui/open-webui#3645] [General]
# Bug Report

## Description

**Bug Summary:**
When using Complete with VSCode, using the apiBase as "https://< url >/ollama/v1" returns an error 404 in the open-webui logs

using "https://< url >/ollama/v1/chat" gives a 500 server error. 

Chat functions in Continue work fine, it's just the completions that give the error. 

**Steps to Reproduce:**
Follow the quick setup guide to install and set up continue in vscode.

## Environment

- **Open WebUI Version:** 0.3.7
- **Ollama (if applicable):** latest

```
  File "/app/backend/main.py", line 523, in dispatch
    prompt = get_last_user_message(data["messages"])
                                   ~~~~^^^^^^^^^^^^
KeyError: 'messages'
INFO:     :0 - "POST /ollama/v1/chat/completions HTTP/1.1" 500 Internal Server Error
```

vscode config:
```
{
  "models": [
    {
      "title": "Deepseek-Coder",
      "model": "deepseek-coder:latest",
      "apiBase": "https://ollama<url>/ollama/v1",
      "apiKey": "sk-<key>",
      "provider": "openai"
    }
  ],
  "customCommands": [
    {
      "name": "test",
      "prompt": "{{{ input }}}\n\nWrite a comprehensive set of unit tests for the selected code. It should setup, run tests that check for correctness including important edge cases, and teardown. Ensure that the tests are complete and sophisticated. Give the tests just as chat output, don't edit any file.",
      "description": "Write unit tests for highlighted code"
    }
  ],
  "tabAutocompleteModel": {
    "title": "Deepseek-Coder",
    "model": "deepseek-coder:latest",
    "apiBase": "https://ollama.<url>/ollama/v1",
    "apiKey": "sk-<key>",
    "provider": "openai"
  }
}
```

**Comment by bdeb1337** [comment]
Hello @jkirkcaldy 

Just like you, I've experienced the same issue trying to link with continue.dev.
When you change your configuration to stop using the legacy completions endpoint, it will work. You can do this by adding `"useLegacyCompletionsEndpoint": false`. This information can be found [here](https://docs.continue.dev/reference/Model%20Providers/openai).

Since the CTRL/CMD+I code generation uses your main model, I would recommended to add it in both pieces of configuration to also make that work.

In your example it would be added as such:

```
{
  "models": [
    {
      "title": "Deepseek-Coder",
      "model": "deepseek-coder:latest",
      "apiBase": "https://<open-webui-url>/ollama/v1",
      "apiKey": "sk-<key>",
      "provider": "openai",
      "useLegacyCompletionsEndpoint": false
    }
  ],
  "customCommands": [
    {
      "name": "test",
      "prompt": "{{{ input }}}\n\nWrite a comprehensive set of unit tests for the selected code. It should setup, run tests that check for correctness including important edge cases, and teardown. Ensure that the tests are complete and sophisticated. Give the tests just as chat output, don't edit any file.",
      "description": "Write unit tests for highlighted code"
    }
  ],
  "tabAutocompleteModel": {
    "title": "Deepseek-Coder",
    "model": "deepseek-coder:latest",
    "apiBase": "https://<open-webui-url>/ollama/v1",
    "apiKey": "sk-<key>",
    "provider": "openai",
    "useLegacyCompletionsEndpoint": false
  }
}
```

To add, my autocomplete with `starcoder2:3b` doesn't really work well yet. Still checking if I'm doing something wrong, but at least the 404 is gone.

**Reply by bdeb1337** [reply]
It's working great for me now. For anyone that might have the same issue: Make sure you **disable** GitHub Copilot extension when testing with Continue. Only disabling the autocomplete was not enough and my Continue autocomplete wouldn't work together. After the disable it works as expected.
